packages/file-explorer/src/FileListContainer.test.tsx

```diff
@@ -0,0 +1,168 @@
+import { TestUtils } from '@deephaven/utils';
+import { ContextMenuRoot } from '@deephaven/components';
+import { FileStorageItem, FileStorageTable } from './FileStorage';
+import FileListContainer, { FileListContainerProps } from './FileListContainer';
+import FileTestUtils from './FileTestUtils';
+
+const renderFileListContainer = async ({
+  table = {} as FileStorageTable,
+  onSelect = jest.fn(),
+  onCopy = undefined,
+  onDelete = undefined,
+  onCreateFile = undefined,
+  onCreateFolder = undefined,
+  onRename = undefined,
+  showContextMenu = false,
+}: Partial<FileListContainerProps>) =>
+  render(
+    <>
+      <FileListContainer
+        table={table}
+        onSelect={onSelect}
+        onCopy={onCopy}
+        onDelete={onDelete}
+        onCreateFile={onCreateFile}
+        onCreateFolder={onCreateFolder}
+        onRename={onRename}
+        showContextMenu={showContextMenu}
+      />
+      <ContextMenuRoot />
+    </>
+  );
+
+it('mounts properly and shows file list', async () => {
+  const files = FileTestUtils.makeFiles();
+  const fileStorage = new MockFileStorage(files);
+  const table = await fileStorage.getTable();
+  renderFileListContainer({ table });
+
+  const foundItems = await screen.findAllByRole('listitem');
+  expect(foundItems).toHaveLength(files.length);
+});
+
+describe('renders correct context menu actions', () => {
+  let user: ReturnType<typeof userEvent.setup>;
+  let files: FileStorageItem[] = [];
+  let table: FileStorageTable;
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  /**
+   * Tests that a button's function is called with expected parameters
+   * @param functionName Name of the function as a string (should match the prop name e.g. onDelete)
+   * @param title Display name of the button
+   * @param expectedValue The parameters the function is expected to receive. If there are multiple, pass them in an array and set multipleExpectedValues to true
+   * @param keyboardText Text you may want to enter after clicking the button
+   * @param multipleExpectedValues True if the function expects multiple parameters
+   */
+  const testContextMenuAction = async (
+    functionName: string,
+    title: RegExp,
+    expectedValue: any,
+    keyboardText?: string,
+    multipleExpectedValues = false
+  ) => {
+    const actionFunction = jest.fn();
+    const props = { table, showContextMenu: true };
+    props[functionName] = actionFunction;
+    renderFileListContainer(props);
+
+    const file = await screen.findByText('testfile1');
+    await TestUtils.rightClick(user, file);
+
+    const button = await screen.findByRole('button', { name: title });
+
+    expect(button).toBeInTheDocument();
+
+    await TestUtils.click(user, button);
+
+    if (keyboardText !== undefined) {
+      await user.keyboard(keyboardText);
+    }
+
+    if (expectedValue !== undefined && multipleExpectedValues) {
+      expect(actionFunction).toHaveBeenCalledWith(...expectedValue);
```

Comment:
Check that `expectedValue` is an array instead of just checking against `undefined`. Then you don't need the `multipleExpectedValues` parameter:
```
if (Array.isArray(expectedValue)) { ... }
```

Reply:
I think the parameter is still needed because onDelete expects an array as an argument without spreading it.